web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Drupal\dpl_loans\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\dpl_react_apps\Controller\DplReactAppsController;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides user loans list.
+ *
+ * @block(
+ *   id = "dpl_loans_list_block",
+ *   admin_label = "List user loans"
+ * )
+ */
+class LoanListBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Drupal config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  private ConfigFactoryInterface $configFactory;
+
+  /**
+   * LoanListBlock constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Drupal config factory to get FBS and Publizon settings.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configuration = $configuration;
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return mixed[]
+   *   The app render array.
+   */
+  public function build() {
+    $context = ['context' => 'Loan list'];
+    $contextAria = ['context' => 'Loan list (Aria)'];
+
+    $fbsConfig = $this->configFactory->get('dpl_fbs.settings');
+    $publizonConfig = $this->configFactory->get('dpl_publizon.settings');
+
+    $data = [
+      // Config.
+      "fbs-base-url" => $fbsConfig->get('base_url'),
+      "publizon-base-url" => $publizonConfig->get('base_url'),
```

Comment:
Should we refactor this to use the same syntax as `DplReactAppsController::searchResultUrl()`?

Reply:
Yes/No. The URL's we get here are external URLs from config. The URL they get in `DplReactAppsController::searchResultUrl()` are internal URLs from routes son not directly comparable.

I think we should have an established way of getting URLs across module boundaries. You could argue that `dpl_loans`should not call an internal(?) function in `dpl_react_apps`.

This should be handled on an architectural level and as such out of scope of this PR.